### ISSUE TITLE
Can't build Ipex from source with compile_bundle.sh for the version v2.1.0+cpu

### ISSUE THREAD

**Issue by JingyaHuang** [intel/intel-extension-for-pytorch#515]
### Describe the bug

Failed to build ipex from source with `compile_bundle.sh` for v2.1.0+cpu (v2.1.1 version works though):

```
/opt/conda/bin/ld: /opt/conda/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2: undefined reference to `_dl_vsym@GLIBC_PRIVATE'

collect2: error: ld returned 1 exit status

make[2]: *** [CMakeFiles/ipex_cpp_test.dir/build.make:117: ipex_cpp_test] Error 1

make[1]: *** [CMakeFiles/Makefile2:136: CMakeFiles/ipex_cpp_test.dir/all] Error 2

make: *** [Makefile:136: all] Error 2

Traceback (most recent call last):
  File "/opt/intel-extension-for-pytorch/setup.py", line 1164, in <module>

    setup(
  File "/opt/conda/lib/python3.10/site-packages/setuptools/__init__.py", line 103, in setup

    return distutils.core.setup(**attrs)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup

    return run_commands(dist)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands

    dist.run_commands()
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands

    self.run_command(cmd)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command

    super().run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command

    cmd_obj.run()
  File "/opt/conda/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 368, in run

    self.run_command("build")
File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command

    self.distribution.run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command

    super().run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command

    cmd_obj.run()
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/command/build.py", line 131, in run

    self.run_command(cmd_name)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command

    self.distribution.run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command

    super().run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command

    cmd_obj.run()
  File "/opt/intel-extension-for-pytorch/setup.py", line 1133, in run

    self.run_command("build_clib")
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command

    self.distribution.run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/dist.py", line 963, in run_command

    super().run_command(command)
  File "/opt/conda/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 988, in run_command

    cmd_obj.run()
  File "/opt/intel-extension-for-pytorch/setup.py", line 835, in run

    _build_project(build_args, get_cpp_test_build_dir(), my_env, use_ninja)
  File "/opt/intel-extension-for-pytorch/setup.py", line 610, in _build_project

    check_call(["make"] + build_args, cwd=build_dir, env=build_env)
  File "/opt/conda/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)

subprocess.CalledProcessError: Command '['make', '-j', '72', 'install']' returned non-zero exit status 2.

^[[91m+ export CXXFLAGS=
+ CXXFLAGS=
+ unset DNNL_GRAPH_BUILD_COMPILER_BACKEND
+ unset LLVM_DIR
+ python -m pip install --force-reinstall 'dist/*.whl'
^[[0m
^[[91mWARNING: Requirement 'dist/*.whl' looks like a filename, but the file does not exist
^[[0m
^[[91mERROR: *.whl is not a valid wheel filename.
^[[0m
The command '/bin/sh -c cd /opt/ && mkdir -p ipex     cd /opt/ipex &&     wget https://raw.githubusercontent.com/intel/intel-extension-for-pytorch/v2.1.0%2Bcpu/scripts/compile_bundle.sh &&     bash compile_bundle.sh &&     /opt/conda/bin/conda install -y jemalloc gperftools -c conda-forge &&     rm -rf /opt/ipex && cd /opt/' returned a non-zero code: 1
```

Also tried with the dockerfile in the ipex repo(`docker/Dockerfile.compile`), and it met the same error.

### Versions

v2.1.0+cpu

(we need this as we are building DLC for AWS services with ipex, and it requires custom PyTorch binaries, and we can't bump to v2.1.1)

**Comment by jgong5** [comment]
To double check, can you dump where the GCC resides in your environment and what is the version of it?

**Comment by jingxu10** [comment]
may i know the command you used to build the docker image with the dockerfile `Dockerfile.compile`?

**Comment by Vasud-ha** [comment]
This issue is reproducible with Dockerfile.compile however building ipex from source with compile.sh script is working fine.

**Comment by Vasud-ha** [comment]
Hi @JingyaHuang, could you please try with the latest ipex v2.1.100+cpu?

**Comment by JingyaHuang** [comment]
Hi @jgong5 @jingxu10, here is the [pull request](https://github.com/aws/deep-learning-containers/pull/3558/files#diff-7349e3410129abfd1cb03037a99f7b076beb5fc48aaefec162101c910cdee185) I opened for integrating Ipex on HF's inference CPU DLC.

@Vasud-ha I can confirm that `compile_bundle.sh` for v2.1.100+cpu works well, and it passed our functional tests as well, however it triggers sanity test failure as for AWS GPU DLC we are opting for v2.1.0 and both images will be tagged v2.1.0 which trigger a tag(v2.1.0) v.s. actual torch version(v2.1.1 after installing ipex v2.1.100) mismatch error.

It would be the best if folks could guide me on install ipex v2.1.0+cpu correctly as we did not find AWS PyTorch binaries for v2.1.1.

**Comment by jingxu10** [comment]
do you need to set it up in a docker env or on a bare metal env?

**Comment by jingxu10** [comment]
Hi @JingyaHuang , is it ok for you to work with PyTorch 2.1.0 + IPEX 2.1.100? I can provide you steps to do so if this is OK.

**Comment by JingyaHuang** [comment]
Hi @jingxu10, thanks for the guide. Actually I am using IPEX v2.1.100+cpu with AWS PyTorch binaries v2.1.0 right now in the PR I mentioned. All functional tests are ok, but since after installing ipex the `torch.__version__==2.1.1`, it caused sanity test failure from the AWS DLC CI which looks like this:
```
>                       assert re.fullmatch(torch_version_pattern, output), (
                            f"torch.__version__ = {output} does not match {torch_version_pattern}\n"
                            f"Please specify framework version as X.Y.Z+cpu"
                        )
E                       AssertionError: torch.__version__ = 2.1.1+cpu does not match 2.1.0(\+cpu)
E                         Please specify framework version as X.Y.Z+cpu
E                       assert None
E                        +  where None = <function fullmatch at 0x7f8ceb351e50>('2.1.0(\\+cpu)', '2.1.1+cpu')
E                        +    where <function fullmatch at 0x7f8ceb351e50> = re.fullmatch
``` 
Does the installation of Ipex impact the version number of torch, and if so when we use ipex v2.1.100 is there any possibility to prevent version number change of pytorch?

**Comment by JingyaHuang** [comment]
@jingxu10 We need to set it up in a containerized environment (AWS Sagemaker Deep Learning Container).

**Comment by ashokei** [comment]
hi @JingyaHuang , your sanity test failure should be resolved by using https://github.com/intel/intel-extension-for-pytorch/issues/515#issuecomment-1908032754

this will replace 2.1.1 expected by ipex with 2.1.0 used by sagemaker container.

can you modify your scripts to add these changes?

**Comment by jingxu10** [comment]
Hi @JingyaHuang , as Ashok mentioned above, the steps I shared in the comment above should fix the CI failure issue.
The reason is in the 2.1.100 `compile_bundle.sh`, we have pytorch 2.1.1 (the latest version when ipex 2.1.100 was released) hard coded. So, the execution of the script installed torch 2.1.1, and conflicted with the CI rule. By applying the steps, the pytorch version will be replaced to 2.1.0 and thus avoid this conflict.

IPEX 2.1.x is compatible with pytorch 2.1.y.
![image](https://github.com/intel/intel-extension-for-pytorch/assets/25193121/cc23e3d9-8fa3-4754-8ef9-e257bcd91623)
